Tomography based mantle convection model cookbook (#6439)
geodynamics/aspect: cookbooks/tomography_based_plate_motions/doc/tomography_based_plate_motions.md

- The reference viscosity profile.
- The prescribed viscosity of the plate boundaries, which controls friction between plates.
- To use viscous and neutrally buoyant cratons in the model.

# Added complexity—Slab2 and initial topography
For additional complexity, we use the Slab2 {cite}`hayes2018slab2` database that describes the three dimensional geometries of all the seismically active subduction zones in detail on Earth instead of the vertical slabs defined in the TM1 model. We also deform the outer shell by the initial topography. Incorporating these finer heterogeneities allow us to simulate Earth-like surface deformation in the models.

> The slabs are included as a compositional field that is diffused with a length scale of 30 km. Physically, this value represents the diffusion of a slab in $\approx$ 15 Ma using diffusivity of 1.14e{-6} m$^2$/s$^{-1}$. Within the slabs, we compute the temperatures using the diffused slabs' composition and a user-defined temperature anomaly, added to the reference mantle adiabat. Additionally, we add a low-viscosity layer in the material model, the mid-mantle layer, that extends below the mantle transition zone until 1000 km depth. The weak mid-mantle layer ensures that our slabs do not get stuck and can still pull the attached lithospheric plates as they sink into the more viscous lower mantle.

Reviewer: Can you add a description for why the slabs are diffused in this model (improve solver stability, more physically realistic, etc)?

For topography, we use a 1-deg spacing longitude-latitude grid and then smoothen the topography using a moving average containing 20 topography data points. The chosen smoothening reduces the pressure oscillations in our models that arise from the sharp topography gradients.

:::{note}

> Our tests with initial topography revealed that having smooth topography variations is important for both linear and nonlinear solver to converge. One possible explanation is that as the outer shell deforms according to the imposed topography, it is more difficult to apply the free-slip boundary condition on a mesh that has sudden variations in the tangential vector across the neighboring cells (see {numref}`fig:topography_test`).

Reviewer: ```suggestion
Our tests with initial topography revealed that having smooth topography variations is important for both linear and nonlinear solver convergence. One possible explanation is that as the outer shell deforms according to the imposed topography, it is more difficult to apply the free-slip boundary condition on a mesh that has sudden variations in the tangential vector across the neighboring cells (see {numref}`fig:topography_test`).
```

- The reference viscosity profile.
- The prescribed viscosity of the plate boundaries, which controls friction between plates.
- To use viscous and neutrally buoyant cratons in the model.

# Added complexity—Slab2 and initial topography

> For additional complexity, we use the Slab2 {cite}`hayes2018slab2` database that describes the three dimensional geometries of all the seismically active subduction zones in detail on Earth instead of the vertical slabs defined in the TM1 model. We also deform the outer shell by the initial topography. Incorporating these finer heterogeneities allow us to simulate Earth-like surface deformation in the models.

Reviewer: ```suggestion
For additional complexity, we use the Slab2 {cite}`hayes2018slab2` database that describes the three dimensional geometries of all the seismically active subduction zones in detail on Earth instead of the vertical slabs defined in the TM1 model. We also deform the outer shell by the initial topography. Incorporating these finer heterogeneities makes the simulations more physically realistic and may improve the fit to observed surface deformation patterns.
```

> The slabs are included as a compositional field that is diffused with a length scale of 30 km. Physically, this value represents the diffusion of a slab in $\approx$ 15 Ma using diffusivity of 1.14e{-6} m$^2$/s$^{-1}$. The diffusion not only approximates the thermal diffusion of the slab as it warms up by the surrounding mantle, but also helps the solver convergence by avoiding sharp transition in material properties between slabs and the surrounding mantle.

Reviewer: ```suggestion
The slabs are included as a compositional field that is diffused with a length scale of 30 km. Physically, this value represents the diffusion of a slab in $\approx$ 15 Ma using a diffusivity of 1.14e{-6} m$^2$/s$^{-1}$. The diffusion not only approximates the thermal diffusion of the slab as it warms up by the surrounding mantle, but also helps the solver convergence by avoiding sharp transition in material properties between slabs and the surrounding mantle.
```

Within the slabs, we compute the temperatures using the diffused slabs' composition and a user-defined temperature anomaly, added to the reference mantle adiabat. Additionally, we add a low-viscosity layer in the material model, the mid-mantle layer, that extends below the mantle transition zone until 1000 km depth. The weak mid-mantle layer ensures that our slabs do not get stuck and can still pull the attached lithospheric plates as they sink into the more viscous lower mantle.

> For topography, we use a 1-deg spacing longitude-latitude grid and then smoothen the topography using a moving average containing 20 topography data points. The chosen smoothening reduces the pressure oscillations in our models that arise from the sharp topography gradients.

Author: Thanks for the catch. The label is correct, I tried several filters. The moving average still works for the 2D case but gives weird pressures in the 3D model and I wanted to be consistent with the 3 models.

```

## Running the model
The prm file that includes all the heterogeneites is available in the folder, [2D_slice_with_faults_slabs_and_topo.prm](https://www.github.com/geodynamics/aspect/blob/main/cookbooks/tomography_based_plate_motions/2D_slice_with_faults_slabs_and_topo.prm). The relevant sections are :

```{literalinclude} tomography_based_plate_motions-2.part.prm
```

The parameter, `Use asthenosphere viscosity scaling in cold regions` ensures that the slabs are uniformly strong and that the low viscosity of asthenosphere does not affect the slabs (i.e, cold regions) during lateral viscosity averaging.

We can visualize the flow field around the Chilean slab in the Figure {numref}`fig:flow_around_slabs` below:

```{figure-md} fig:flow_around_slabs
<img src="Fig_flow_around_slabs.*" />

> Instantaneous velocity field as Chilean slab subducts and pulls the attached Nazca plate, and the trench suction acting on the overriding South American plate. The low-viscosity layer added above the slab is to represent the weak sediment layer above a slab. The dashed white line represents the base of the low mid-mantle viscosity that allows the slabs to subduct freely below the upper mantle.

Reviewer: ```suggestion
Instantaneous velocity field illustrating subduction of the Chilean slab, slab pull on the attached Nazca plate, and resulting the trench suction acting on the overriding South American plate. The low-viscosity layer added above the slab is an approximation of the weak sediment layer above a slab. The dashed white line represents the base of the low mid-mantle viscosity that allows the slabs to subduct freely below the upper mantle.
```

```

## Surface deformation : plate motions and SHmax directions
With access to high-performance computing, we would want to run these models in three-dimensions and compare the modeled surface observations such as plate motions or SHmax directions.

> The following model was run using ~2.5k cores on a HPC and shows a practical application of how these models can be used to approximate the present-day physical state of the Earth.

Reviewer: ```suggestion
The following model was run using ~2.5k cores on the NSF-funded High Performance Computing System (Frontera) at the University of Texas, which shows a practical application of how these models can be used to approximate the present-day physical state of the Earth.
```

- To use viscous and neutrally buoyant cratons in the model.

# Added complexity—Slab2 and initial topography
For additional complexity, we use the Slab2 {cite}`hayes2018slab2` database that describes in detail the three dimensional geometries
of all the seismically active subduction zones on Earth instead of the vertical slabs defined in the TM1 model. We also deform the outer
shell by the initial topography. Incorporating these finer heterogeneities makes the simulations more physically realistic and may improve
the fit to observed surface deformation patterns.

The slabs are included as a compositional field that is diffused with a length scale of 30 km. Physically, this value represents the

> diffusion of a slab in $\approx$ 15 Ma using diffusivity of $1.14\times10^{-6}$. The diffusion not only approximates the thermal diffusion

Reviewer 2: I think you removed the unit from the diffusivity.

solver convergence. One possible explanation is that as the outer shell deforms according to the imposed topography, it is more
difficult to apply the free-slip boundary condition on a mesh that has sudden variations in the tangential vector across the
neighboring cells (see {numref}`fig:topography_test`).
:::

```{figure-md} fig:topography_test
<img src="Fig_topography_test.*" />

(a) Flow in a simple compressible spherical shell with added topography—a mountain and a basin. The arrows marks the region where the
tangential flow changes from deformed to undeformed surface. Shorter-wavelength topography changes leads to higher linear iterations
and higher non-linear residual in the system. (b) We smooth the observed topography in this cookbook because the material model is quite
complex already for sufficient solver convergence.
```

## Running the model
The prm file that includes all the heterogeneites is available in the folder,
[2D_slice_with_faults_slabs_and_topo.prm](https://www.github.com/geodynamics/aspect/blob/main/cookbooks/tomography_based_plate_motions/2D_slice_with_faults_slabs_and_topo.prm).
The relevant sections are :

> ```{literalinclude} tomography_based_plate_motions-2.part.prm

Reviewer 2: Do you need to update this file now that you added new input parameters?
